src/util/bitvector_expr.cpp

```diff
@@ -142,3 +142,17 @@ exprt count_trailing_zeros_exprt::lower() const
 
   return typecast_exprt::conditional_cast(result, type());
 }
+
+exprt bitreverse_exprt::lower() const
+{
+  const std::size_t int_width = to_bitvector_type(type()).get_width();
+
+  exprt::operandst result_bits;
+  result_bits.reserve(int_width);
+
+  const exprt operand = op();
+  for(std::size_t i = 0; i < int_width; ++i)
+    result_bits.push_back(extractbit_exprt{operand, i});
+
```

Comment:
This is potentially problematic when the lowered expression is passed to any piece of code that doesn't cope well with the sharing. This is worth at least a warning, and it may possibly make sense to fix this issue by introducing a `let_exprt` for `op()`.

Reply:
I've made a stab at it - is the second commit that I added what you were looking for?

Reply:
yes!